sleap_roots/points.py

```diff
@@ -139,13 +139,13 @@ def get_all_pts_array(pts0: np.ndarray, *args: Optional[np.ndarray]) -> np.ndarr
     return np.concatenate(concatenated_pts, axis=0)
 
 
-def get_nodes(pts: np.ndarray, node_index: int) -> np.ndarray:
+def get_nodes(pts: np.ndarray, node_index: int | np.ndarray) -> np.ndarray:
```

Comment:
can we simplify the logic here a bit?

first validate type, then for each type have strict requirements. 

Why is 0.0 singled-out? what about 1.0 etc.?

Make sure each conditional has a corresponding test in tests